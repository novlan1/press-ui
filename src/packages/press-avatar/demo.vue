<template>
  <div class="demo-wrap">
    <demo-block :title="t('basicUsage')">
      <PressAvatar :src="src" />
    </demo-block>

    <demo-block :title="t('shape')">
      <PressAvatar
        :src="src"
        round
        :custom-style="customStyle"
      />
      <PressAvatar
        :src="src"
        :round="false"
      />
    </demo-block>

    <demo-block :title="t('size')">
      <PressAvatar
        :src="src"
        size="30"
        :custom-style="customStyle"
      />
      <PressAvatar
        :src="src"
        size="40"
        :custom-style="customStyle"
      />
      <PressAvatar
        :src="src"
        size="50"
      />
    </demo-block>

    <demo-block :title="t('icon')">
      <PressAvatar
        :src="src"
        icon="star-o"
        :custom-style="customStyle"
      />
      <PressAvatar
        :src="src"
        icon="like-o"
      />
    </demo-block>

    <demo-block :title="t('text')">
      <PressAvatar
        text="Y"
        :custom-style="customStyle"
      />
      <PressAvatar
        text="G"
        random-background-color
        :custom-style="customStyle"
      />
      <PressAvatar
        text="杨"
        random-background-color
      />
    </demo-block>

    <demo-block :title="t('error')">
      <PressAvatar />
    </demo-block>

    <demo-block :title="t('avatarGroup')">
      <PressAvatarGroup
        :urls="urls"
        custom-style="margin-bottom: 16px"
        @showMore="showMore"
      />

      <PressAvatarGroup
        :urls="urls"
        gap="0.6"
        custom-style="margin-bottom: 16px"
        @showMore="showMore"
      />

      <PressAvatarGroup
        :urls="urls"
        size="50"
        custom-style="margin-bottom: 16px"
        @showMore="showMore"
      />
    </demo-block>
  </div>
</template>
<script>
import PressAvatarGroup from 'press-ui/press-avatar-group/press-avatar-group.vue';
import PressAvatar from 'press-ui/press-avatar/press-avatar.vue';


const src = 'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_0ab8e66284a3617ed9.jpeg';


export default {
  i18n: {
    'zh-CN': {
      shape: '头像形状',
      size: '头像尺寸',
      icon: '图标头像',
      text: '文字头像',
      error: '加载失败',
      avatarGroup: '头像组',
    },
    'en-US': {
      shape: 'Shape',
      size: 'Size',
      icon: 'Icon',
      text: 'Text',
      error: 'Error',
      avatarGroup: 'Avatar Group',
    },
  },
  components: {
    PressAvatar,
    PressAvatarGroup,
  },
  data() {
    return {
      src,
      customStyle: 'margin-right: 20px',
      urls: [
        src,
        'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_ef8e6956717db34745.jpg',
        src,
        'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_7157200e469d0468b3.jpg',
        src,
        'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/10/own_mike_a17f94292a34e9f644.jpeg',
        src,
      ],
    };
  },
  methods: {
    showMore() {
      this.onGTip('[showMore]');
    },
  },
};
</script>
<style scoped lang="scss">
</style>
