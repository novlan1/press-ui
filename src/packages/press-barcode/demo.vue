<template>
  <div class="demo-wrap demo-wrap--gray">
    <div class="barcode-wrap">
      <PressBarcode
        :value="value"
        :width="options.width"
        :height="options.height"
        :format="options.format"
        :font-size="options.fontSize"
      />
    </div>

    <DemoInput
      :autosize="autosize"
      :value="value"
      @confirm="onConfirmInput"
    />
  </div>
</template>
<script>
import PressBarcode from 'press-ui/press-barcode/press-barcode.vue';
import DemoInput from 'press-ui/press-demo-input/press-demo-input.vue';


export default {
  components: {
    PressBarcode,
    DemoInput,
  },
  data() {
    return {
      autosize: { maxHeight: 600, minHeight: 50 },
      value: '123123123123',
      options: {
        width: 1,
        height: 50,
        fontSize: 12,
        format: 'CODE128',
      },
    };
  },
  methods: {
    onConfirmInput(value) {
      this.value = value;
    },
  },
};
</script>
<style scoped lang="scss">
.barcode-wrap {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
}
</style>
