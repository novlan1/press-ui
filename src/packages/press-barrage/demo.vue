<template>
  <div class="demo-wrap">
    <demo-block
      :title="t('basicUsage')"
    >
      <PressBarrage v-model="list">
        <div
          class="video"
          style="width: 100%; height: 150px; background: #f2f3f5;"
        />
      </PressBarrage>

      <PressButton
        type="primary"
        size="small"
        custom-style="margin-top: 10px;"
        @click="add"
      >
        弹幕
      </PressButton>
    </demo-block>

    <demo-block
      :title="t('videoBarrage')"
    >
      <PressBarrage
        ref="pressBarrage"
        v-model="videoList"
        :auto-play="false"
      >
        <div
          class="video"
          style="width: 100%; height: 150px; background: #f2f3f5;"
        />
      </PressBarrage>

      <div>
        <PressButton
          type="primary"
          size="small"
          :disabled="!isPlay"
          custom-style="margin-top: 10px;margin-right: 10px;"
          @click="videoAdd"
        >
          弹幕
        </PressButton>

        <PressButton
          type="default"
          size="small"
          custom-style="margin-top: 10px;"
          @click="toggle"
        >
          {{ isPlay ? t('pause') : t('play') }}
        </PressButton>
      </div>
    </demo-block>
  </div>
</template>
<script>
import { setAdapter } from 'press-ui/common/vue3/set';
import PressBarrage from 'press-ui/press-barrage/press-barrage.vue';
import PressButton from 'press-ui/press-button/press-button.vue';


export default {
  i18n: {
    'zh-CN': {
      barrage: '弹幕',
      play: '开始',
      pause: '暂停',
      videoBarrage: '模仿视频弹幕',
      lightweight: '轻量',
      customizable: '可定制的',
      mobile: '移动端',
      library: '组件库',
    },
    'en-US': {
      barrage: 'barrage',
      play: 'play',
      pause: 'pause',
      videoBarrage: 'Imitate video barrage',
      lightweight: 'Lightweight',
      customizable: 'Customizable',
      mobile: 'Mobile',
      library: 'Library',
    },
  },
  components: {
    PressBarrage,
    PressButton,
  },
  data() {
    const defaultList = [
      { id: 100, text: this.t('lightweight') },
      { id: 101, text: this.t('customizable') },
      { id: 102, text: this.t('mobile') },
      { id: 103, text: 'Vue' },
      { id: 104, text: this.t('library') },
      { id: 105, text: 'PressUI' },
      { id: 106, text: '666' },
    ];

    return {
      list: defaultList,
      videoList: [
        ...defaultList,
      ],
      isPlay: false,

    };
  },
  watch: {
    isPlay: {
      handler(val) {
        if (val) {
          this.$refs.pressBarrage?.play();
        } else {
          this.$refs.pressBarrage?.pause();
        }
      },
    },
  },
  methods: {
    add() {
      setAdapter(this, 'list', [
        ...this.list,
        {
          id: Math.random(),
          text: 'Barrage',
        },
      ]);
    },
    videoAdd() {
      setAdapter(this, 'videoList', [
        ...this.videoList,
        {
          id: Math.random(),
          text: 'Barrage',
        },
      ]);
    },
    toggle() {
      this.isPlay = !this.isPlay;
    },
  },
};
</script>
<style scoped lang="scss">
</style>
