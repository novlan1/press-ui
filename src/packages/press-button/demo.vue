<template>
  <div class="demo-wrap">
    <demo-block :title="t('type')">
      <div class="demo-button-row">
        <PressButton
          type="default"
          :custom-style="customStyle"
          @click="onClick"
        >
          {{ t('default') }}
        </PressButton>
        <PressButton
          type="primary"
          :custom-style="`${customStyle}`"
          @click="onVibrateShort"
        >
          {{ t('primary') }}
        </PressButton>
        <PressButton
          type="info"
          @click="onVibrateLong"
        >
          {{ t('info') }}
        </PressButton>
      </div>

      <div class="demo-button-row">
        <PressButton
          type="warning"
          :custom-style="customStyle"
          @click="onLaunchMp"
        >
          {{ t('warning') }}
        </PressButton>
        <PressButton
          type="danger"
          @click="onLaunchMp(1)"
        >
          {{ t('danger') }}
        </PressButton>
      </div>
    </demo-block>

    <demo-block :title="t('plain')">
      <div class="demo-button-row">
        <PressButton
          plain
          type="default"
          :custom-style="customStyle"
        >
          {{ t('plain') }}
        </PressButton>
        <PressButton
          plain
          type="primary"
          :custom-style="customStyle"
        >
          {{ t('plain') }}
        </PressButton>
      </div>

      <PressButton
        plain
        type="info"
      >
        {{ t('plain') }}
      </PressButton>
    </demo-block>


    <demo-block :title="t('hairline')">
      <PressButton
        plain
        hairline
        type="primary"
        :custom-style="customStyle"
      >
        {{ t('hairlineButton') }}
      </PressButton>
      <PressButton
        plain
        hairline
        type="info"
      >
        {{ t('hairlineButton') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('disabled')">
      <div class="demo-button-row">
        <PressButton
          disabled
          type="primary"
          :custom-style="customStyle"
        >
          {{ t('disabled') }}
        </PressButton>
        <PressButton
          disabled
          type="info"
          :custom-style="customStyle"
        >
          {{ t('disabled') }}
        </PressButton>
      </div>

      <PressButton
        disabled
        type="primary"
        :custom-style="customStyle"
      >
        {{ t('disabled') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('loading')">
      <PressButton
        loading
        type="primary"
        :custom-style="customStyle"
      />
      <PressButton
        loading
        type="primary"
        loading-type="spinner"
        :custom-style="customStyle"
      />
      <PressButton
        loading
        loading-type="circular-tdesign"
        type="info"
        :loading-text="t('loadingText')"
      />
    </demo-block>

    <demo-block :title="t('shape')">
      <PressButton
        square
        type="primary"
        :custom-style="customStyle"
      >
        {{ t('square') }}
      </PressButton>
      <PressButton
        round
        type="info"
      >
        {{ t('round') }}
      </PressButton>
    </demo-block>


    <demo-block :title="t('icon')">
      <PressButton
        icon="star-o"
        type="primary"
        :custom-style="customStyle"
      />
      <PressButton
        icon="star-o"
        type="primary"
        :custom-style="customStyle"
      >
        {{ t('button') }}
      </PressButton>
      <PressButton
        icon="https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/press%2Fimg%2Fpress-ui-avatar-transparent-2.png"
      >
        {{ t('button') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('size')">
      <div class="demo-button-row">
        <PressButton
          type="primary"
          size="large"
        >
          {{ t('large') }}
        </PressButton>
      </div>

      <PressButton
        type="primary"
        size="normal"
        :custom-style="customStyle"
      >
        {{ t('normal') }}
      </PressButton>
      <PressButton
        type="primary"
        size="small"
        :custom-style="customStyle"
      >
        {{ t('small') }}
      </PressButton>
      <PressButton
        type="primary"
        size="mini"
      >
        {{ t('mini') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('blockElement')">
      <PressButton
        type="primary"
        block
      >
        {{ t('blockElement') }}
      </PressButton>
    </demo-block>

    <demo-block :title="t('customColor')">
      <PressButton
        color="#7232dd"
        :custom-style="customStyle"
        open-type="openSetting"
      >
        {{ t('pure') }}
      </PressButton>
      <PressButton
        color="#7232dd"
        plain
        :custom-style="customStyle"
        open-type="feedback"
      >
        {{ t('pure') }}
      </PressButton>
      <PressButton
        color="linear-gradient(to right, #4bb0ff, #6149f6)"
        open-type="share"
      >
        {{ t('gradient') }}
      </PressButton>
    </demo-block>

    <demo-block
      v-for="(item,index) of [0,1,2]"
      :key="index"
      :title="index===1 ?'E-SPORT-DISABLE' : index===2 ? 'E-SPORT-LOADING' : 'E-SPORT'"
    >
      <div class="demo-button-row">
        <!-- :size="{width: '136px', height: '36px'}" -->
        <PressButton
          type="e-sport-primary"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
        >
          primary
        </PressButton>
        <PressButton
          type="e-sport-primary-bg"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
        >
          primary-bg
        </PressButton>
      </div>

      <div class="demo-button-row">
        <PressButton
          type="e-sport-primary-bg-lg"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
        >
          primary-bg-lg
        </PressButton>
      </div>

      <div class="demo-button-row">
        <PressButton
          type="e-sport-primary-bg-xl"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
        >
          primary-bg-xl
        </PressButton>
      </div>

      <div class="demo-button-row">
        <PressButton
          type="e-sport-secondary"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
          :size="{width: '136px', height: '36px'}"
        >
          secondary
        </PressButton>

        <!-- 线框的loader图片是透明的，没效果，直接隐藏 -->
        <PressButton
          v-if="index!==2"
          type="e-sport-border"
          :disabled="index===1"
          :loading="index===2"
          :custom-style="customStyle"
          :size="{width: '136px', height: '36px'}"
        >
          border
        </PressButton>
      </div>
    </demo-block>
  </div>
</template>
<script>
import PressButton from 'press-ui/press-button/press-button.vue';
import { isNotInUni } from 'press-ui/common/utils/utils';
import { launchMiniProgramInGame } from 't-comm/lib/msdk/mini-program';


export default {
  i18n: {
    'zh-CN': {
      type: '按钮类型',
      size: '按钮尺寸',
      icon: '图标按钮',
      loading: '加载状态',
      disabled: '禁用状态',
      shape: '按钮形状',
      default: '默认按钮',
      primary: '主要按钮',
      info: '信息按钮',
      danger: '危险按钮',
      warning: '警告按钮',
      large: '大号按钮',
      normal: '普通按钮',
      small: '小型按钮',
      mini: '迷你按钮',
      plain: '朴素按钮',
      square: '方形按钮',
      round: '圆形按钮',
      hairline: '细边框',
      hairlineButton: '细边框按钮',
      loadingText: '加载中...',
      router: '页面导航',
      urlRoute: 'URL 跳转',
      vueRoute: '路由跳转',
      customColor: '自定义颜色',
      pure: '单色按钮',
      gradient: '渐变色按钮',
      blockElement: '块级元素',
    },
    'en-US': {
      type: 'Type',
      size: 'Size',
      icon: 'Icon',
      loading: 'Loading',
      disabled: 'Disabled',
      shape: 'Shape',
      default: 'Default',
      primary: 'Primary',
      info: 'Info',
      danger: 'Danger',
      warning: 'Warning',
      large: 'Large',
      normal: 'Normal',
      small: 'Small',
      mini: 'Mini',
      plain: 'Plain',
      square: 'Square',
      round: 'Round',
      hairline: 'Hairline',
      hairlineButton: 'Hairline',
      loadingText: 'Loading...',
      router: 'Router',
      urlRoute: 'URL',
      vueRoute: 'Vue Router',
      customColor: 'Custom Color',
      pure: 'Pure',
      gradient: 'Gradient',
      blockElement: 'Block Element',
    },
  },
  components: {
    PressButton,
  },
  data() {
    return {
      customStyle: 'margin-right: 16px;',
    };
  },
  onReady() {

  },
  methods: {
    onLaunchMp(isQQ) {
      console.log('[launchMiniProgramInGame]');
      // #ifdef H5
      launchMiniProgramInGame({
        appId: 'wx99c1c10a389e7433',
        isWxMp: !isQQ,
      });
      // #endif
    },
    onClick(event) {
      console.log('[CLICK]', event);
    },
    onVibrateLong() {
      if (isNotInUni()) return;

      uni.vibrateLong({
        success() {
          console.log('success');
        },
        fail() {
          console.log('fail');
        },
      });
    },
    onVibrateShort() {
      // #ifdef H5
      navigator.vibrate(1000);
      // #endif

      if (isNotInUni()) return;

      uni.vibrateLong({
        success() {
          console.log('success');
        },
        fail() {
          console.log('fail');
        },
      });
    },
  },
};
</script>

<style scoped lang="scss">
.demo-button-row {
  margin-bottom: 12px;
}
</style>

