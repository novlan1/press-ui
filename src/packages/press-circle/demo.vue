<template>
  <div class="demo-wrap demo-wrap--gray">
    <demo-block
      :title="t('basicUsage')"
      :header-style="headerStyle"
      :section-style="sectionStyle"
    >
      <PressCircle
        :value="value"
        :text="`${value}%`"
      />
    </demo-block>

    <demo-block
      :title="t('customStyle')"
      :header-style="headerStyle"
      :section-style="sectionStyle"
    >
      <div class="demo-part">
        <PressCircle
          :value="value"
          :stroke-width="6"
          :text="t('customWidth')"
        />
      </div>

      <div class="demo-part">
        <PressCircle
          :value="value"
          :stroke-width="6"
          layer-color="#eeeeee"
          color="#ee0a24"
          :text="t('customColor')"
        />
      </div>

      <div class="demo-part">
        <PressCircle
          :value="value"
          :stroke-width="6"
          :color="gradientColor"
          :text="t('gradient')"
        />
      </div>

      <div class="demo-part">
        <PressCircle
          :value="value"
          :stroke-width="6"
          color="#07c160"
          :clockwise="false"
          :text="t('counterClockwise')"
        />
      </div>

      <div class="demo-part">
        <PressCircle
          :value="value"
          :size="120"
          :text="t('customSize')"
        />
      </div>

      <div>
        <PressButton
          type="e-sport-primary"
          custom-style="margin-right:8px;"
          @click="onChangeCircleValue('plus')"
        >
          {{ t('add') }}
        </PressButton>
        <PressButton
          type="e-sport-primary"
          @click="onChangeCircleValue('minus')"
        >
          {{ t('decrease') }}
        </PressButton>
      </div>
    </demo-block>
  </div>
</template>
<script>
import PressButton from 'press-ui/press-button/press-button.vue';
import PressCircle from 'press-ui/press-circle/press-circle.vue';


export default {
  i18n: {
    'zh-CN': {
      gradient: '渐变色',
      customSize: '大小定制',
      customStyle: '样式定制',
      customColor: '颜色定制',
      customWidth: '宽度定制',
      counterClockwise: '逆时针',
    },
    'en-US': {
      gradient: 'Gradient',
      customSize: 'Size',
      customStyle: 'Custom Style',
      customColor: 'Color',
      customWidth: 'Width',
      counterClockwise: 'Direction',
    },
  },

  components: {
    PressCircle,
    PressButton,
  },
  data() {
    return {
      value: 20,
      headerStyle: 'background: #f7f8fa;',
      sectionStyle: 'background: #f7f8fa;margin: 0;padding: 0 12px;',
      gradientColor: {
        '0%': '#3fecff',
        '100%': '#6149f6',
      },
    };
  },
  methods: {
    onChangeCircleValue(type) {
      const STEP = 20;

      if (type === 'plus') {
        if (this.value >= 100) return;
        this.value += STEP;
      } else if (type === 'minus') {
        if (this.value <= 0) return;
        this.value -= STEP;
      }
    },
  },
};
</script>
<style lang="scss" scoped>
.demo-part {
  margin: 0 12px 12px 0;
  display: inline-block;
}
</style>
