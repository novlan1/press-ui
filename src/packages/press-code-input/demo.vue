<template>
  <div class="demo-wrap">
    <demo-block :title="t('basicUsage')">
      <PressCodeInput
        v-model="value"
        @change="change"
        @finish="finish"
      />
    </demo-block>

    <demo-block :title="t('mode')">
      <PressCodeInput
        v-model="valueMode"
        mode="line"
      />
    </demo-block>

    <demo-block :title="t('maxLength')">
      <PressCodeInput
        v-model="valueMaxLength"
        :max-length="4"
      />
    </demo-block>

    <demo-block :title="t('space')">
      <PressCodeInput
        v-model="valueSpace"
        :space="0"
        :max-length="4"
      />
    </demo-block>

    <demo-block :title="t('hairline')">
      <PressCodeInput
        v-model="valueHairline"
        hairline
        :max-length="4"
      />

      <!-- <PressCodeInput
        v-model="valueHairline2"
        mode="line"
        hairline
      /> -->
    </demo-block>

    <demo-block :title="t('style')">
      <PressCodeInput
        v-model="valueStyle"
        color="#f56c6c"
        border-color="#f56c6c"
        :max-length="4"
      />
    </demo-block>

    <demo-block :title="t('dot')">
      <PressCodeInput
        v-model="valueDot"
        dot
        :max-length="4"
      />
    </demo-block>

    <demo-block :title="t('autoFocus')">
      <PressCodeInput
        v-model="valueFocus"
        focus
        :max-length="4"
      />
    </demo-block>
  </div>
</template>
<script>
import PressCodeInput from 'press-ui/press-code-input/press-code-input.vue';

export default {
  i18n: {
    'zh-CN': {
      mode: '模式',
      maxLength: '自定义长度',
      space: '自定义间距',
      hairline: '细边框',
      style: '自定义样式',
      dot: '隐藏输入内容',
      autoFocus: '自动获取焦点',
    },
    'en-US': {
      mode: 'Mode',
      maxLength: 'Max Length',
      space: 'Space',
      hairline: 'Hairline',
      style: 'Custom Style',
      dot: 'Dot',
      autoFocus: 'Auto Focus',
    },
  },
  components: {
    PressCodeInput,
  },
  data() {
    return {
      value: '',
      valueMode: '',
      valueSpace: '',
      valueMaxLength: '',
      valueHairline: '',
      valueHairline2: '',
      valueStyle: '',
      valueDot: '',
      valueFocus: '',
    };
  },
  methods: {
    change(e) {
      console.log(`内容改变，当前值为：${e}`);
    },
    finish(e) {
      console.log(`输入结束，当前值为：${e}`);
    },
  },
};
</script>
<style scoped lang="scss">
.demo-wrap {
  padding-bottom: 30px;
}
</style>
