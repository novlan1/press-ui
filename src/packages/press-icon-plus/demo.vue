<template>
  <div class="demo-wrap demo-wrap--gray">
    <!-- #ifndef H5 -->
    <PressNotify
      id="press-notify"
    />
    <!-- #endif -->

    <PressTabs
      :offset-top="offsetTop"
      sticky
      swipeable
    >
      <PressTab :title="t('demo')">
        <DemoBlockPlus :title="t('basicUsage')">
          <PressCol
            :span="6"
          >
            <PressIconPlus
              :name="demoIcon"
              @click="copy(demoIcon)"
            />
          </PressCol>
          <PressCol
            :span="6"
          >
            <PressIconPlus
              :name="demoImage"
              @click="copy(demoImage)"
            />
          </PressCol>
        </DemoBlockPlus>

        <DemoBlockPlus :title="t('badge')">
          <PressCol
            :span="6"
          >
            <PressIconPlus
              :name="demoIcon"
              dot
              @click="copy(demoIcon, { dot: true })"
            />
          </PressCol>
          <PressCol
            :span="6"
          >
            <PressIconPlus
              :name="demoIcon"
              info="9"
              @click="copy(demoIcon, { badge: '9' })"
            />
          </PressCol>
          <PressCol
            :span="6"
          >
            <PressIconPlus
              :name="demoIcon"
              info="99+"
              @click="copy(demoIcon, { badge: '99+' })"
            />
          </PressCol>
        </DemoBlockPlus>

        <DemoBlockPlus :title="t('color')">
          <PressCol
            :span="6"
          >
            <PressIconPlus
              name="cart-o"
              color="#1989fa"
              @click="copy('cart-o', { color: '#1989fa' })"
            />
          </PressCol>
          <PressCol
            :span="6"
          >
            <PressIconPlus
              name="fire-o"
              :color="RED"
              @click="copy('fire-o', { color: RED })"
            />
          </PressCol>
        </DemoBlockPlus>

        <DemoBlockPlus :title="t('size')">
          <PressCol
            :span="6"
          >
            <PressIconPlus
              :name="demoIcon"
              size="40"
              @click="copy(demoIcon, { size: '40' })"
            />
          </PressCol>
          <PressCol
            :span="6"
          >
            <PressIconPlus
              :name="demoIcon"
              size="1rem"
              @click="copy(demoIcon, { size: '3rem' })"
            />
          </PressCol>
        </DemoBlockPlus>
      </PressTab>

      <PressTab :title="t('basic')">
        <PressCol
          v-for="(icon) in icons.basic"
          :key="icon"
          :span="6"
        >
          <PressIconPlus
            :name="icon"
            @click="copy(icon)"
          />
          <span>{{ icon }}</span>
        </PressCol>
      </PressTab>

      <PressTab :title="t('outline')">
        <PressCol
          v-for="icon in icons.outline"
          :key="icon"
          :span="6"
        >
          <PressIconPlus
            :name="icon"
            @click="copy(icon)"
          />
          <span>{{ icon }}</span>
        </PressCol>
      </PressTab>

      <PressTab :title="t('filled')">
        <PressCol
          v-for="icon in icons.filled"
          :key="icon"
          :span="6"
        >
          <PressIconPlus
            :name="icon"
            @click="copy(icon)"
          />
          <span>{{ icon }}</span>
        </PressCol>
      </PressTab>
    </PressTabs>
  </div>
</template>

<script>
import { RED } from 'press-ui/common/constant/color';
import PressCol from 'press-ui/press-col/press-col.vue';
import DemoBlockPlus from 'press-ui/press-demo-block-plus/press-demo-block-plus.vue';

import icons from 'press-ui/press-icon-plus/config';
import PressIconPlus from 'press-ui/press-icon-plus/press-icon-plus.vue';

import PressNotify from 'press-ui/press-notify/press-notify.vue';
import PressTab from 'press-ui/press-tab/press-tab.vue';
import PressTabs from 'press-ui/press-tabs/press-tabs.vue';

import { CLIPBOARD_MIXIN } from 'src/packages/press-icon-plus/demo-helper/clipboard-mixin';


export default {
  i18n: {
    'zh-CN': {
      copied: '复制成功',
      title: '图标列表',
      badge: '徽标提示',
      basic: '基础图标',
      outline: '线框风格',
      filled: '实底风格',
      demo: '用法示例',
      color: '图标颜色',
      size: '图标大小',
    },
    'en-US': {
      copied: 'Copied',
      title: 'Icon List',
      badge: 'Show Badge',
      basic: 'Basic',
      outline: 'Outline',
      filled: 'Filled',
      demo: 'Demo',
      color: 'Icon Color',
      size: 'Icon Size',
    },
  },
  components: {
    PressTabs,
    PressTab,
    PressCol,
    PressIconPlus,
    PressNotify,
    DemoBlockPlus,
  },
  mixins: [CLIPBOARD_MIXIN],
  options: {
    styleIsolation: 'shared',
  },
  data() {
    return {
      RED,
      demoIcon: 'chat-o',
      demoImage: 'https://mike-1255355338.cos.ap-guangzhou.myqcloud.com/article/2023/5/own_mike_ce77489af93cb34c4b.png',

      icons: {
        basic: [],
        outline: [],
        filled: [],
      },
    };
  },
  mounted() {
    setTimeout(() => {
      this.icons = {
        ...(this.icons),
        ...(icons || {}),
      };
    }, 1000);
  },
  methods: {
    copy(icon, option = {}) {
      let tag = `<press-icon-plus name="${icon}"`;
      if ('dot' in option) {
        tag = `${tag} ${option.dot ? 'dot' : ''}`;
      }
      if ('badge' in option) {
        tag = `${tag} badge="${option.badge}"`;
      }
      if ('color' in option) {
        tag = `${tag} color="${option.color}"`;
      }
      if ('size' in option) {
        tag = `${tag} size="${option.size}"`;
      }
      tag = `${tag} />`;

      console.log('[tag] ', tag);
      this.copyIconTag(tag);
    },
  },
};
</script>

<style scoped lang="scss">
@import "press-ui/common/style/press/var.scss";

.demo-wrap {
  ::v-deep .press-tab__pane {
    width: auto;
    margin: 20px;
    background-color: #fff;
    border-radius: 12px;
  }

  ::v-deep .press-tabs__content {
    background-color: #f7f8fa;
  }

  ::v-deep .press-col {
    float: none !important;
    text-align: center;
    vertical-align: middle;
    cursor: pointer;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;

    span {
      display: block;
      height: 36px;
      margin: -4px 0 4px;
      padding: 0 5px;
      color: $gray-7;
      font-size: 12px;
      line-height: 18px;
    }

    &:active {
      background-color: $active-color;
    }
  }

  ::v-deep .press-icon-plus {
    margin: 16px 0 16px;
    color: $text-color;
    font-size: 32px;
  }
}
</style>
