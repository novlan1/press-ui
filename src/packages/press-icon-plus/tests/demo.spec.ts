import { snapshotDemo } from '../../../../tests/unit/demo';
import Demo from '../demo.vue';

jest.mock('t-comm/es/clipboard/index', () => ({
  clipboardMp: () => {},
  clipboardWeb: () => {},
}));

snapshotDemo(Demo);
