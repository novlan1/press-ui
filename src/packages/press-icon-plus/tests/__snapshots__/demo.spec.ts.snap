// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo-wrap">
  <div class="press-notify-index" id="press-notify">
    <!---->
  </div>
  <div class="press-card press-card--full press-card--border" style="margin: 0px; padding: 0px 10px;">
    <!---->
    <!---->
    <div class="press-card__content" style="padding: 10px;"><span class="uni-h6">
      仅保留 empty 图标，其他图标请使用 press-icon-plus。
    </span></div>
    <div class="press-card__actions"></div>
  </div>
  <div>
    <div class="icon-content">
      <div class="icon-item"><span class="iconfont icon-empty" style="font-size: 30px;"></span> <span class="uni-mt-5 uni-subtitle" style="color: rgb(94, 109, 130);">
          empty
        </span></div>
    </div>
  </div>
  <div>
    <div class="icon-content flex-end">
      <div class="icon-item">
        <div class="press-icon-music" style="width: 40px; height: 50px;">
          <div class="press-icon-music__bar press-icon-music__bar-0" style="background-color: rgb(41, 121, 255); width: calc(40px / 4);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-1" style="background-color: rgb(41, 121, 255); width: calc(40px / 4);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-2" style="background-color: rgb(41, 121, 255); width: calc(40px / 4);"></div>
        </div>
      </div>
      <div class="icon-item">
        <div class="press-icon-music" style="width: 40px; height: 50px;">
          <div class="press-icon-music__bar press-icon-music__bar-0" style="background-color: rgb(94, 109, 130); width: calc(40px / 4);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-1" style="background-color: rgb(94, 109, 130); width: calc(40px / 4);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-2" style="background-color: rgb(94, 109, 130); width: calc(40px / 4);"></div>
        </div>
      </div>
      <div class="icon-item">
        <div class="press-icon-music" style="width: 40px; height: 50px;">
          <div class="press-icon-music__bar press-icon-music__bar-0" style="background-color: rgb(41, 121, 255); width: calc(40px / 6);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-1" style="background-color: rgb(41, 121, 255); width: calc(40px / 6);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-2" style="background-color: rgb(41, 121, 255); width: calc(40px / 6);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-3" style="background-color: rgb(41, 121, 255); width: calc(40px / 6);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-4" style="background-color: rgb(41, 121, 255); width: calc(40px / 6);"></div>
        </div>
      </div>
      <div class="icon-item">
        <div class="press-icon-music" style="width: 60px; height: 60px;">
          <div class="press-icon-music__bar press-icon-music__bar-0" style="background-color: rgb(41, 121, 255); width: calc(60px / 4);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-1" style="background-color: rgb(41, 121, 255); width: calc(60px / 4);"></div>
          <div class="press-icon-music__bar press-icon-music__bar-2" style="background-color: rgb(41, 121, 255); width: calc(60px / 4);"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
